### PR TITLE
Proto Query Interface

### PR DESCRIPTION
**Pull request by TobiasHafner**
This pull request introduces a new query interface plugin called the Proto Interface, designed to enhance Polypheny's capability by enabling multimodel querying through a protobuf and gRPC-based protocol.

### Changes

- **Architecture Expansion**: A new interface plugin was added, and a new protobuf and gRPC-based protocol implemented.

### Features

- **Proto Interface Plugin**: A new plugin facilitating multimodel querying capabilities across different data models using protobuf and gRPC.

### Tests

- **Integration Tests**: Implemented to ensure the Proto Interface correctly interacts with existing Polypheny components.
- **Performance Tests**: Conducted to verify that the introduction of the new interface does not adversely affect the system's performance.

### Notice

The plugin is not enabled, and the interface is not deployed by default. This will happen as part of a second pull request that introduces additional improvements. Further, this new interface will, as part of this next PR, also be renamed to Prism-Interface.

### CODE REVIEW

File: dbms/src/main/java/org/polypheny/db/processing/AbstractQueryProcessor.java [hunk 1]
```diff
@@ -1068,6 +1068,7 @@ private PreparedResult<PolyValue> implement( AlgRoot root, AlgDataType parameter
         }
 
         final AlgDataType jdbcType = QueryProcessorHelpers.makeStruct( root.alg.getCluster().getTypeFactory(), root.validatedRowType );
+        //TODO DL: get metadata instead of null
```
Comment (vogti) [inline dbms/src/main/java/org/polypheny/db/processing/AbstractQueryProcessor.java:1071]: ToDo?

File: dbms/src/main/java/org/polypheny/db/transaction/StatementImpl.java [hunk 2]
```diff
@@ -173,7 +173,7 @@ public void close() {
             dataContext.getParameterValues().clear();
         }
         fileInputHandles.forEach( FileInputHandle::close );
-        dataContext = null;
+        // dataContext = null; todo dl check how we want to handle this long term
```
Comment (vogti) [inline dbms/src/main/java/org/polypheny/db/transaction/StatementImpl.java:176]: ToDo?

File: dbms/src/main/java/org/polypheny/db/processing/AbstractQueryProcessor.java (hunk 1 above)
Comment (datomo) [reply dbms/src/main/java/org/polypheny/db/processing/AbstractQueryProcessor.java:1071]: I added this comment to mark the next line, as we are not able to provide the `fieldOrigins `information and always use `null`. This is correct for the one current use in this method but can be misleading if the variable is used at another place. I now moved the variable and added a comment.

File: dbms/src/main/java/org/polypheny/db/transaction/StatementImpl.java (hunk 2 above)
Comment (datomo) [reply dbms/src/main/java/org/polypheny/db/transaction/StatementImpl.java:176]: Removed. Handling should hold up to our current capabilities and use-cases.

**Review by datomo — COMMENTED** [review]
Thank you @TobiasHafner, this looks good.

**Review by vogti — APPROVED** [review]
Thx, @TobiasHafner, for this PR!

**Pull request merged**